File: content/snippets/python/language_basics/loops.py
#!/usr/bin/env python3

def main() -> None:
    values = [1, 2, 3, 4, 5]

    for i in range(len(values)):
        print("for index:", i, "->", values[i])

    for v in values:
        if v == 3:
            continue
        print("for each:", v)

    i = 0
    while i < len(values):
        if values[i] == 4:
            break
        print("while:", values[i])
        i += 1

    c = 0
    while True:
        print("do-while style iteration:", c)
        c += 1
        if c >= 2:
            break


if __name__ == "__main__":
    main()

File: content/snippets/typescript/language_basics/loops.ts
#!/usr/bin/env ts-node

import { setImmediate } from "node:timers";

function main(): void {
  const values = [1, 2, 3, 4, 5];

  for (let i = 0; i < values.length; i++) {
    console.log("for index:", i, "->", values[i]);
  }

  for (const v of values) {
    if (v === 3) {
      continue;
    }
    console.log("for...of:", v);
  }

  let i = 0;
  while (i < values.length) {
    if (values[i] === 4) {
      break;
    }
    console.log("while:", values[i]);
    i++;
  }

  let c = 0;
  do {
    console.log("do...while iteration:", c);
    c++;
  } while (c < 2);
}

main();
setImmediate(() => {});
